generate-java.sh

```diff
@@ -2,6 +2,7 @@
 
 dir_resolve()
 {
+"$1"!="" && mkdir -p "$1"
```

Comment:
Any particular reason to make the target optional for the Java generator? Seems to me we'd like parity with the Go generator; both or neither should make the target optional.